dimod/include/dimod/quadratic_model.h

```diff
@@ -252,9 +254,52 @@ class Neighborhood {
         }
     }
 
+    /// Request that the neighborhood capacity be at least enough to contain `n`
+    /// elements.
+    void reserve(index_type n) {
+        neighbors.reserve(n);
+        quadratic_biases.reserve(n);
+    }
+
     /// Return the size of the neighborhood.
     size_type size() const { return neighbors.size(); }
 
+    /// Sort the neighborhood and sum the biases of duplicate variables.
+    void sort_and_sum() {
```

Comment:
Do you want to move this to the utils like the zip sort one ? Up to you.

Reply:
Eh, for now I'll leave it be and move it when it becomes relevant.